streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -182,7 +188,7 @@ public void writeAsText(String filePath, Serde<K> keySerde, Serde<V> valSerde) {
         try {
 
             PrintStream printStream = new PrintStream(new FileOutputStream(filePath));
-            topology.addProcessor(name, new KeyValuePrinter<>(printStream, keySerde, valSerde), this.name);
+            topology.addProcessor(name, new KeyValuePrinter<>(printStream, keySerde, valSerde, null), this.name);
```

Comment:
> Should we extend (ie, new overloads) writeAsText with optional parameter name similar to print() ?

@mjsax 

I'm on the fence with this one.  I thought about adding the overloaded methods for `writeAsText()`, but decided not to.  The `print()` function writes to Stdout. With multiple streams the results are intermingled and steam/ktable names help with disambiguation.  The `writeAsText()` function writes the output of each stream/ktable to it's own file, so, in my opinion, having names for the stream/ktable doesn't add much.  

What do you think?